### ISSUE TITLE
Support libmpv2?

### ISSUE THREAD

**Issue by florianduclot** [olivierfriard/BORIS#547]
I wasn't sure whether to file this as a _bug report_ or a _feature request_; my apologies if this isn't the right place.

**Describe the bug**
I just tried running from source on my Tumbleweed system and, as expected, ran into a few compatibility issues due to the fact that this system is running relatively recent versions of some dependencies. I thought I would list the main one here as a heads-up in case it hasn't been brought up before (I couldn't find a mention of it in this repo).
The first was a failure to `pip install numpy` when running Python3.10, but that was easily resolved by running in a `virtualenv` based on python3.8.
The main incompatibility I wanted to report here is related to `libmpv`. Indeed, when trying to start `boris`, I encountered:
```
python3 -m boris
Traceback (most recent call last):
  File "/usr/lib64/python3.8/runpy.py", line 185, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib64/python3.8/runpy.py", line 144, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/usr/lib64/python3.8/runpy.py", line 111, in _get_module_details
    __import__(pkg_name)
  File "/home/florian/Documents/Tools/boris/boris_venv/lib/python3.8/site-packages/boris/__init__.py", line 24, in <module>
    from .core import main
  File "/home/florian/Documents/Tools/boris/boris_venv/lib/python3.8/site-packages/boris/core.py", line 97, in <module>
    from boris import player_dock_widget
  File "/home/florian/Documents/Tools/boris/boris_venv/lib/python3.8/site-packages/boris/player_dock_widget.py", line 28, in <module>
    from boris import mpv
  File "/home/florian/Documents/Tools/boris/boris_venv/lib/python3.8/site-packages/boris/mpv.py", line 531, in <module>
    _handle_func('mpv_detach_destroy',          [],                                         None, errcheck=None)
  File "/home/florian/Documents/Tools/boris/boris_venv/lib/python3.8/site-packages/boris/mpv.py", line 478, in _handle_func
    func = getattr(backend, name)
  File "/usr/lib64/python3.8/ctypes/__init__.py", line 386, in __getattr__
    func = self.__getitem__(name)
  File "/usr/lib64/python3.8/ctypes/__init__.py", line 391, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
AttributeError: /lib64/libmpv.so.2: undefined symbol: mpv_detach_destroy
```
As you can see above, my system is running mpv-2. Details are:
```
# zypper info libmpv2
Loading repository data...
Reading installed packages...


Information for package libmpv2:
--------------------------------
Repository     : openSUSE:Tumbleweed
Name           : libmpv2
Version        : 0.34.1+git.20220203T182214.5d31c7310c-2.3
Arch           : x86_64
Vendor         : openSUSE
Installed Size : 2.9 MiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : mpv-0.34.1+git.20220203T182214.5d31c7310c-2.3.src
Upstream URL   : http://mpv.io
Summary        : A library to link together with mpv player
Description    : 
    mpv is a movie player based on MPlayer and mplayer2. It supports a wide variety
    of video file formats, audio and video codecs, and subtitle types.

    This package contains a library that can other apps use to utilize the mpv
    features.
```

After looking at the current `mpv.py` from upstream (https://github.com/jaseg/python-mpv/blob/30d5ca1f0255184b500cbb43d9b3556a70038c53/mpv.py), I noticed that `mpv_detach_destroy` is indeed not present anymore in mpv-2.
Replacing `mpv.py` in my boris install with upstream `mpv.py` linked above resolves the launch issue.

Would it be safe or useful to consider switching to libmpv2 for the upcoming v8?

Thanks a lot for any feedback.

**To Reproduce**
```
python3 -m boris
```

**Expected behavior**
Boris successfully starts.

**Screenshots**
See error log above.

**Desktop (please complete the following information):**
 - OS: openSUSE Tumbleweed 20221013
 - BORIS Version: version 8.0.14 release date: BETA 2022-02-24

**Comment by olivierfriard** [comment]
Hi,

thank you for reporting this.
The current version of BORIS (8.9.2) available on PyPI and GitHub (mpv branch)  should run with libmpv v.2.
I tested it on OpenSuse Tumbleweed and Windows 10 VMs.

I am not able to reproduce the problem you obtained with `pip install numpy` with Python 3.10.

Here are the details of my Python 3.10.7 virtual environment:

```
Package         Version
--------------- -------
boris-behav-obs 8.9.2
contourpy       1.0.5
cycler          0.11.0
defusedxml      0.7.1
et-xmlfile      1.1.0
ExifRead        3.0.0
fonttools       4.38.0
kiwisolver      1.4.4
MarkupPy        1.14
matplotlib      3.6.1
numpy           1.23.4
odfpy           1.4.1
openpyxl        3.0.10
packaging       21.3
pandas          1.3.5
Pillow          9.2.0
pip             22.3
pyparsing       3.0.9
PyQt5           5.15.7
PyQt5-Qt5       5.15.2
PyQt5-sip       12.11.0
pyreadr         0.4.7
python-dateutil 2.8.2
pytz            2022.5
setuptools      63.2.0
six             1.16.0
tablib          3.2.1
tabulate        0.9.0
xlrd            2.0.1
xlwt            1.3.0


```

**Comment by florianduclot** [comment]
>  The current version of BORIS (8.9.2) available on PyPI and GitHub (mpv branch) should run with libmpv v.2.

Thanks a lot for the added support of mpv2, it's greatly appreciated! I did not notice the `mpv` branch before so thanks for pointing that out, too.

> I am not able to reproduce the problem you obtained with pip install numpy with Python 3.10.

I just tried again and it cannot reproduce my previous failure anymore... not sure what happened here but I'm glad all is well. I tried again by creating a new py3.10.7 virtualenv and I get the same lib versions as you after installing boris from PyPi. Sorry for the distraction there.

Feel free to close if you'd like, and thanks again.
